tests/testthat/test-dup.R

```diff
@@ -13,3 +13,21 @@ test_that("0 duplicates equals empty string", {
 test_that("uses tidyverse recycling rules", {
   expect_error(str_dup(1:2, 1:3), class = "vctrs_error_incompatible_size")
 })
+
+test_that("uses sep argument", {
+  expect_equal(str_dup("abc", 1, sep = "-"), "abc")
```

Comment:
I tried to make it a bit more clear was different in each test by keeping more stuff the same.